realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -1288,15 +1292,1274 @@ public void add(@Nullable E e) {
+        }
+
+        final int size = size();
+        final float[] result = new float[size];
+        for (int i = 0; i < size; i++) {
+            final E element = osListOperator.get(i);
+            result[i] = (element == null) ? valueForNull : ((Number) element).floatValue();
+        }
+        return result;
+    }
+
+    /**
+     * Returns newly allocated {@code float} array that contains all values in this {@link RealmList}.
+     * <p>
+     * This method is available only when the list is managed and the element type is implementing {@link Number} interface.
+     * <p>
+     * If this list contains {@code null} value, {@code 0F} is used instead.
+     *
+     * @return newly allocated {@code float} array.
+     * @throws IllegalStateException if this {@link RealmList} is not managed or element type is not implementing {@link Number}.
+     * @see #toFloatArray(float)
+     * @see #toArray()
+     * @see #toArray(Object[])
+     */
+    public float[] toFloatArray() {
+        return toFloatArray(0F);
+    }
+
+    private static boolean isClassForRealmModel(Class<?> clazz) {
+        return RealmModel.class.isAssignableFrom(clazz);
+    }
+
+    private ManagedListOperator<E> getOperator(BaseRealm realm, OsList osList, @Nullable Class<E> clazz, @Nullable String className) {
+        if (clazz == null || isClassForRealmModel(clazz)) {
+            return new RealmModelListOperator<>(realm, osList, clazz, className);
+        }
+        if (clazz == String.class) {
+            //noinspection unchecked
+            return (ManagedListOperator<E>) new StringListOperator(realm, osList, (Class<String>) clazz);
+        }
+        if (clazz == Long.class || clazz == Integer.class || clazz == Short.class || clazz == Byte.class) {
+            return new LongListOperator<>(realm, osList, clazz);
+        }
+        if (clazz == Boolean.class) {
+            //noinspection unchecked
+            return (ManagedListOperator<E>) new BooleanListOperator(realm, osList, (Class<Boolean>) clazz);
+        }
+        if (clazz == byte[].class) {
+            //noinspection unchecked
+            return (ManagedListOperator<E>) new BinaryListOperator(realm, osList, (Class<byte[]>) clazz);
+        }
+        if (clazz == Double.class) {
+            //noinspection unchecked
+            return (ManagedListOperator<E>) new DoubleListOperator(realm, osList, (Class<Double>) clazz);
+        }
+        if (clazz == Float.class) {
+            //noinspection unchecked
+            return (ManagedListOperator<E>) new FloatListOperator(realm, osList, (Class<Float>) clazz);
+        }
+        if (clazz == Date.class) {
+            //noinspection unchecked
+            return (ManagedListOperator<E>) new DateListOperator(realm, osList, (Class<Date>) clazz);
+        }
+        throw new IllegalArgumentException("Unexpected value class: " + clazz.getName());
+    }
+}
+
+abstract class ManagedListOperator<T> {
```

Comment:
Please add Javadoc describing what this inner class is for

Reply:
added